pkg/virt-operator/install-strategy/create.go

```diff
@@ -0,0 +1,995 @@
+			}
+			ops = append(ops, labelAnnotationPatch...)
+
+			_, err = core.ServiceAccounts(kv.Namespace).Patch(sa.Name, types.JSONPatchType, generatePatchBytes(ops))
+			if err != nil {
+				return false, fmt.Errorf("unable to patch serviceaccount %+v: %v", sa, err)
+			}
+			log.Log.V(2).Infof("serviceaccount %v updated", sa.GetName())
+
+		} else {
+			// Up to date
+			log.Log.V(2).Infof("serviceaccount %v already exists and is up-to-date", sa.GetName())
+		}
+	}
+
+	// create/update ClusterRoles
+	for _, cr := range targetStrategy.clusterRoles {
+		var cachedCr *rbacv1.ClusterRole
+
+		cr := cr.DeepCopy()
+		obj, exists, _ := stores.ClusterRoleCache.Get(cr)
+
+		if exists {
+			cachedCr = obj.(*rbacv1.ClusterRole)
+		}
+
+		injectOperatorLabelAndAnnotations(&cr.ObjectMeta, imageTag, imageRegistry)
+		if !exists {
+			// Create non existent
+			expectations.ClusterRole.RaiseExpectations(kvkey, 1, 0)
+			_, err := rbac.ClusterRoles().Create(cr)
+			if err != nil {
+				expectations.ClusterRole.LowerExpectations(kvkey, 1, 0)
+				return false, fmt.Errorf("unable to create clusterrole %+v: %v", cr, err)
+			}
+			log.Log.V(2).Infof("clusterrole %v created", cr.GetName())
+		} else if !objectMatchesVersion(&cachedCr.ObjectMeta, imageTag, imageRegistry) {
+			// Update existing, we don't need to patch for rbac rules.
+			_, err = rbac.ClusterRoles().Update(cr)
+			if err != nil {
+				return false, fmt.Errorf("unable to update clusterrole %+v: %v", cr, err)
+			}
+			log.Log.V(2).Infof("clusterrole %v updated", cr.GetName())
+
+		} else {
+			log.Log.V(4).Infof("clusterrole %v already exists", cr.GetName())
+		}
+	}
+
+	// create/update ClusterRoleBindings
+	for _, crb := range targetStrategy.clusterRoleBindings {
+
+		var cachedCrb *rbacv1.ClusterRoleBinding
+
+		crb := crb.DeepCopy()
+		obj, exists, _ := stores.ClusterRoleBindingCache.Get(crb)
+		if exists {
+			cachedCrb = obj.(*rbacv1.ClusterRoleBinding)
+		}
+
+		injectOperatorLabelAndAnnotations(&crb.ObjectMeta, imageTag, imageRegistry)
+		if !exists {
+			// Create non existent
+			expectations.ClusterRoleBinding.RaiseExpectations(kvkey, 1, 0)
+			_, err := rbac.ClusterRoleBindings().Create(crb)
+			if err != nil {
+				expectations.ClusterRoleBinding.LowerExpectations(kvkey, 1, 0)
+				return false, fmt.Errorf("unable to create clusterrolebinding %+v: %v", crb, err)
+			}
+			log.Log.V(2).Infof("clusterrolebinding %v created", crb.GetName())
+		} else if !objectMatchesVersion(&cachedCrb.ObjectMeta, imageTag, imageRegistry) {
+			// Update existing, we don't need to patch for rbac rules.
+			_, err = rbac.ClusterRoleBindings().Update(crb)
+			if err != nil {
+				return false, fmt.Errorf("unable to update clusterrolebinding %+v: %v", crb, err)
+			}
+			log.Log.V(2).Infof("clusterrolebinding %v updated", crb.GetName())
+
+		} else {
+			log.Log.V(4).Infof("clusterrolebinding %v already exists", crb.GetName())
+		}
+	}
+
+	// create/update Roles
+	for _, r := range targetStrategy.roles {
```

Comment:
Same here for updates, they may have to exist in parallel, instead of doing direct updates.